### ISSUE TITLE
Unfortunately, an unexpected error occurred, and Bundler cannot continue.

### ISSUE THREAD

**Issue by smohanty-mdsol** [rubygems/rubygems#4644]
### Getting the below error after running bundle install
Unfortunately, an unexpected error occurred, and Bundler cannot continue.

### Which command did you run?
bundle install

### What were you expecting to happen?

It should install all the gems and create the gemfile.lock file.

### What actually happened?
Given the below error while running bundle.
`Unfortunately, an unexpected error occurred, and Bundler cannot continue.`

### If not included with the output of your command, run `bundle env` and paste the output below

<img width="1440" alt="Screen Shot 2021-06-02 at 10 43 04 AM" src="https://user-images.githubusercontent.com/26060849/120427673-f7333500-c38f-11eb-9411-972de383f7fb.png">

**Comment by duckinator** [comment]
I'm very confused why the error manifested this way when you did `bundle env`, but if you take the URL in the screenshot and decode it, it says:

```
Permission denied @ rb_sysopen - /Users/smohanty/.rvm/gems/ruby-2.7.0/gems/mysql2-0.4.10/CHANGELOG.md
```

And I would not be surprised if a permissions error got you a generic "an unexpected error occurred" message, so I'm wondering if that might be the problem.

Can you run the following commands and provide the output for each?

* `stat /Users/smohanty/.rvm/gems/ruby-2.7.0/gems`
* `stat /Users/smohanty/.rvm/gems/ruby-2.7.0`
* `stat /Users/smohanty/.rvm/gems/`
* `stat /Users/smohanty/.rvm/`

**Comment by smohanty-mdsol** [comment]
_**Please find the below responses for the above commands respectivelly.**_
C02SP7KYG8WN:mccadmin smohanty$ **stat /Users/smohanty/.rvm/gems/ruby-2.7.0/gems**
`16777220 13131718543 drwxr-xr-x 778 smohanty (1362359299 0 24896 "Jun  7 21:41:59 2021" "Jun  7 21:40:13 2021" "Jun  7 21:40:13 2021" "May  4 00:38:25 2020" 4096 0 0 /Users/smohanty/.rvm/gems/ruby-2.7.0/gems`
C02SP7KYG8WN:mccadmin smohanty$ **stat /Users/smohanty/.rvm/gems/ruby-2.7.0**
`16777220 13131715118 drwxr-xr-x 13 smohanty (1362359299 0 416 "Jun  7 21:42:06 2021" "Jun  7 21:40:30 2021" "Jun  7 21:40:30 2021" "May  4 00:14:26 2020" 4096 0 0 /Users/smohanty/.rvm/gems/ruby-2.7.0`
C02SP7KYG8WN:mccadmin smohanty$ **stat /Users/smohanty/.rvm/gems/**
`16777220 898603 drwxr-xr-x 54 (777) (1362359299 0 1728 "Jun  2 13:50:08 2021" "Jun  2 13:47:07 2021" "Jun  2 20:03:02 2021" "Feb 27 16:45:36 2017" 4096 0 0 /Users/smohanty/.rvm/gems/`
C02SP7KYG8WN:mccadmin smohanty$ **stat /Users/smohanty/.rvm/**
`16777220 897745 drwxr-xr-x 33 smohanty (1362359299 0 1056 "Mar 29 17:19:34 2021" "Aug 28 23:33:49 2019" "Aug 28 23:33:49 2019" "Feb 27 16:45:20 2017" 4096 0 0 /Users/smohanty/.rvm/`

**Comment by smohanty-mdsol** [comment]
This issue has been resolved with ruby version 2.7.2, hence closing this issue.
